eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,151 @@
+# Description: This script is used to verify the specs used to generate SDK package point to the main branch of Azure/azure-rest-api-specs repository
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $ServiceDirectory,
+  [Parameter(Position = 1)]
+  [ValidateNotNullOrEmpty()]
+  [string] $PackageName
+)
+
+. $PSScriptRoot/common.ps1
+. $PSScriptRoot/Helpers/PSModule-Helpers.ps1
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+
+# This function is used to verify the 'require' and 'input-file' settings in autorest.md point to the main branch of Azure/azure-rest-api-specs repository
+function Verify-AutorestMd([string]$fileUrl) {
+  if($fileUrl -match "^https://github.com/(?<repo>[^/]*/azure-rest-api-specs)/blob/(?<commit>[0-9a-f]{40})/(?<path>.*)") {
+    $repo = $matches['repo']
+    $commit = $matches['commit']
+    if($repo -ne "Azure/azure-rest-api-specs") {
+      LogError "Invalid repo in the file url: $fileUrl. Repo should be 'Azure/azure-rest-api-specs'."
+      exit 1
+    }
+    # check the commit hash belongs to main branch
+    Verify-CommitFromMainBranch $commit
+  }
+  else{
+    LogError "Invalid file url: $fileUrl"
+    exit 1
+  }
+}
+
+# This function is used to verify the 'repo' and 'commit' settings in tsp-location.yaml point to the main branch of Azure/azure-rest-api-specs repository
+function Verify-TspLocation([System.Object]$tspLocationObj) {
+  $repo = $tspLocationObj["repo"]
+  $commit = $tspLocationObj["commit"]
+  if($repo -ne "Azure/azure-rest-api-specs") {
+    LogError "Invalid repo setting in the tsp-location.yaml: $repo. Repo should be 'Azure/azure-rest-api-specs'."
+    exit 1
+  }
+
+  # check the commit hash belongs to main branch
+  Verify-CommitFromMainBranch $commit
+}
+
+# This function is used to verify the specific 'commit' belongs to the main branch of Azure/azure-rest-api-specs repository
+function Verify-CommitFromMainBranch([string]$commit) {
+  $mainBranch = "main"
+  $specRepoCloneDir = "./tmp_spec_repo"
+  New-Item $specRepoCloneDir -Type Directory -Force | Out-Null
+  Push-Location $specRepoCloneDir
+
+  try {
+    $repoRemoteUrl = Get-GitRemoteValue "Azure/azure-rest-api-specs"
+    git clone -b main $repoRemoteUrl .
```

Comment:
Cloning down the specs repo in this check seems pretty heavy handed. Is there a github api we can call to check if a it is part of main? Or perhaps we could use the sparse clone from the generation command? We could even potentially add this as a check in the generate command.

Reply:
Perhaps you could use the search api as outlined in https://stackoverflow.com/questions/30376787/github-api-check-if-a-branch-or-repository-contains-a-commit

Reply:
Example:
`https://api.github.com/search/commits?q=repo:Azure/azure-sdk-tools+hash:dff04c39ef72f91c27cb10f164b10b6bf958edc9`

Reply:
Right, I also thought that git clone is a little bit heavier. I looked at this search api previously but hesitated to use it concerning the paging behavior. I think we will not hit the paging case for current use case.